homeassistant/components/lifx/coordinator.py

```diff
@@ -110,6 +110,58 @@ def current_infrared_brightness(self) -> str | None:
         """Return the current infrared brightness as a string."""
         return infrared_brightness_value_to_option(self.device.infrared_brightness)
 
+    async def diagnostics(self) -> dict[str, Any]:
+        """Return diagnostic information about the device."""
+        features = lifx_features(self.device)
+        device_data = {
+            "firmware": self.device.host_firmware_version,
+            "version": {
+                "vendor": self.device.vendor,
+                "product_id": self.device.product,
+            },
+            "features": features,
+            "state": {
+                "hue": round(self.device.color[0] / 65535 * 360, 2),
+                "saturation": round(self.device.color[1] / 65535 * 100, 2),
+                "brightness": round(self.device.color[2] / 65535 * 100, 2),
+                "kelvin": self.device.color[3],
+                "power": "on" if self.device.power_level == 65535 else "off",
```

Comment:
Isn't the point of diagnostics to communicate raw values?

Reply:
Yes, we want the raw data so we can use it to debug.